homeassistant/components/qnap_qsw/__init__.py

```diff
@@ -0,0 +1,77 @@
+"""The QNAP QSW integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from aioqsw.const import QSD_FIRMWARE, QSD_FIRMWARE_INFO, QSD_PRODUCT, QSD_SYSTEM_BOARD
+from aioqsw.localapi import ConnectionOptions, QnapQswApi
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_URL, CONF_USERNAME, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import aiohttp_client
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, MANUFACTURER
+from .coordinator import QswUpdateCoordinator
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+
+class QswEntity(CoordinatorEntity[QswUpdateCoordinator]):
+    """Define an QNAP QSW entity."""
+
+    def __init__(
+        self,
+        coordinator: QswUpdateCoordinator,
+        entry: ConfigEntry,
+    ) -> None:
+        """Initialize."""
+        super().__init__(coordinator)
+
+        self._attr_device_info: DeviceInfo = {
+            "configuration_url": entry.data[CONF_URL],
+            "identifiers": {(DOMAIN, entry.entry_id)},
+            "manufacturer": MANUFACTURER,
+            "model": self.get_entity_value(QSD_SYSTEM_BOARD, QSD_PRODUCT),
+            "name": self.get_entity_value(QSD_SYSTEM_BOARD, QSD_PRODUCT),
+            "sw_version": self.get_entity_value(QSD_FIRMWARE_INFO, QSD_FIRMWARE),
+        }
```

Comment:
```suggestion
        self._attr_device_info = DeviceInfo(
            configuration_url=entry.data[CONF_URL],
            identifiers={(DOMAIN, entry.entry_id)},
            manufacturer=MANUFACTURER,
            model=self.get_entity_value(QSD_SYSTEM_BOARD, QSD_PRODUCT),
            name=self.get_entity_value(QSD_SYSTEM_BOARD, QSD_PRODUCT),
            sw_version=self.get_entity_value(QSD_FIRMWARE_INFO, QSD_FIRMWARE)
        )
```

We should also use `connections` instead of `identifiers` since we have the mac

Reply:
Changed in ccf3a61013a649c3bb9521ac2d1d1ee67e2ff281